### PR TITLE
Add support for negative augment indexes

### LINKED ISSUE #2352: Allow negative offsets for `mat`'s `augment` argument.
### Description

It would count from the end, so
```
mat(augment: -2,
  a, b, c;
  d, e, f;
  g, h, i;
)
```
would be equivalent to `augment: 1`.
(This would of course apply to the other forms of the argument as well)

### Use Case

Consider for example a definition for a Matrix defining a system of equations:
```
#let eqs-mat = mat.with(augment: -1)
```

This would be difficult to implement currently, requiring to inquire into the column count of the matrix.

### PR DESCRIPTION
**Pull request by SekoiaTree**
Fixes #2352.

Not *in love* with throwing conversions to isize here and there, but there's no real way to avoid it. This would become a problem if anybody makes a matrix with more than 4294967295 columns or rows, but I think that's already beyond reasonable.

### CODE REVIEW

**Comment by laurmaedje** [comment]
I think we should resolve the augment indices in one place from signed to unsigned, instead of having it across all usages. See, for instance `locate{_opt}` in `Array` for an example.

File: crates/typst-library/src/math/matrix.rs [hunk 1]
```diff
@@ -200,27 +202,34 @@ impl LayoutMath for MatElem {
...
 
         let augment = self.augment(ctx.styles());
+        let rows = self.rows();
 
         if let Some(aug) = &augment {
             for &offset in &aug.hline.0 {
-                if offset == 0 || offset >= self.rows().len() {
+                if offset == 0
+                    || offset >= (rows.len() as isize)
+                    || offset <= -(rows.len() as isize)
```
Comment (EpicEricEE) [inline crates/typst-library/src/math/matrix.rs:211]: This (and the same thing below) could be
```rs
if offset == 0 || offset.unsigned_abs() >= rows.len()
```

Comment (SekoiaTree) [reply crates/typst-library/src/math/matrix.rs]: Good catch, thanks!

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**